crates/miden-objects/src/account/storage/map/partial.rs

```diff
@@ -21,15 +23,17 @@ use crate::account::{StorageMap, StorageMapWitness};
 #[derive(Clone, Debug, PartialEq, Eq, Default)]
 pub struct PartialStorageMap {
     partial_smt: PartialSmt,
+    map: BTreeMap<Word, Word>,
 }
 
 impl PartialStorageMap {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns a new instance of partial storage map with the specified partial SMT.
-    pub fn new(partial_smt: PartialSmt) -> Self {
-        PartialStorageMap { partial_smt }
+    /// Returns a new instance of partial storage map with the specified partial SMT and stored
+    /// entries.
+    pub fn new(partial_smt: PartialSmt, map: BTreeMap<Word, Word>) -> Self {
```

Comment:
Would it work to do what @bobbinth suggested in another comment:

```rust
pub struct StorageMapWitness {
    raw_keys: Vec<Word>,
    proof: SmtProof,
}
```

and then change this to `PartialStorageMap::from_proofs(proofs: impl IntoIterator<Item = StorageMapWitness>)`? That would be a nicer API and avoid the need to validate that the map has all the keys that the partial SMT tracks.